src/b-splines/cubic.jl

```diff
@@ -0,0 +1,235 @@
+"""
+Assuming uniform knots with spacing 1, the `i`th piece of cubic spline
+implemented here is defined as follows.
+
+    y_i(x) = ϕm p(x-i) + ϕ q(x-i) + ϕp q(1- (x-i)) + ϕpp p(1 - (x-i))
+
+where
+
+    p(☆) = 1/6 *(1-☆)^3
+    q(☆) = 2/3 - ☆^2 + 1/2 ☆^3
+
+and the values `ϕX` for `X ⋹ {m, _, p, pp}` are the pre-filtered coefficients.
+
+For future reference, this expands out to the following polynomial:
+
+    y_i(x) = 1/6 cm (1+i-x)^3 + c (2/3 - (x-i)^2 + 1/2 (x-i)^3) +
+             cp (2/3 - (1+i-x)^2 + 1/2 (1+i-x)^3) + 1/6 cpp (x-i)^3
+
+When we derive boundary conditions we will use derivatives `y_0'(x)` and
+`y_0''(x)`
+"""
+Cubic
```

Comment:
I'm definitely a fan of moving documentation this way - it keeps it with the implementation and makes it easy to access. Nicely done!